db/system_keyspace.cc

```diff
@@ -2276,7 +2276,10 @@ class runtime_info_table : public memtable_filling_virtual_table {
         add_partition(mutation_sink, "trace_probability", format("{:.2}", tracing::tracing::get_local_tracing_instance().get_trace_probability()));
         co_await add_partition(mutation_sink, "memory", [this] () {
             struct stats {
-                uint64_t total = 0;
+                // take the pre-reserved memory into account, as seastar only returns
+                // the stats of memory managed by the seastar allocator, but we instruct
+                // it to reserve addition memory for system.
+                uint64_t total = db::config::wasm_udf_reserved_memory;
```

Comment:
Is it reserved per shard?

Reply:
indeed. will fix.

Reply:
I'm confused. If total is passed to map_reduce so it would be summed smp::count times, so I think the initialization was correct.

My concern was that the 50MB value was for all shards (like the totla memory option is)

Reply:
sorry, i misinterpreted https://github.com/scylladb/seastar/blob/b5865a96039243c74db8d1e21f21099dde3028fe/src/core/resource.cc#L512 . 50MB is reserved from **total** memory, i.e., the "1G" in `--memory 1G`.

Reply:
>  If total is passed to map_reduce so it would be summed smp::count times, so I think the initialization was correct.

no, how so? it will be something like
```py
total = db::config::wasm_udf_reserved_memory + stats_on_shard[0].total + stats_on_shard[1].total + ...
```

so the "total" is correct. i am not sure if i am following you now.

Reply:
math is hard ;)
I trust your calculation.